grammar.js

```diff
@@ -121,11 +121,12 @@ module.exports = grammar({
       repeat($._blank),
     ),
 
-    // Codeblock: preformatted block of lines starting with ">".
+    // Codeblock: preformatted block of lines starting with ">", followed by optional language.
     codeblock: ($) => prec.right(seq(
       '>',
+      optional(alias(token.immediate(/[a-z0-9]+/), $.language)),
       token.immediate('\n'),
-      repeat1(alias($.line_code, $.line)),
+      alias(repeat1(alias($.line_code, $.line)), $.code),
```

Comment:
is this for semantics or does it have a functional purpose? Maybe `$.codeline` fits better with the existing naming patterns.

Reply:
It's necessary (as far as I can tell) so the whole block gets punted to the injected parser; you can't parse, e.g.,
```lua
for k,v, in pairs(foo) do
  print(v)
end
```
purely line by line

Reply:
To be clear, this aliases the whole set of repeated lines as a single node, not each single line.

Reply:
But that's what `(codeblock)` is for.

Reply:
Because that includes the markers, which are _not_ part of the injected code.

Reply:
(But I'm not married to the name, of course; I just need a single node that I can capture as `@contents`; see the `injections.scm` query.)

Reply:
> Because that includes the markers, which are not part of the injected code.

I want to believe that's solvable because they're anonymous :) Any hope?

Reply:
Not unless you give me some ;) Honestly, it would help me if you could explain the nature of the objection to doing it the way I was able to.